application/utxohandler/utxotrie/utxotrie_test.go

```diff
@@ -173,3 +187,2457 @@ func TestUTXOTrie(t *testing.T) {
 
 }
 */
+
+func TestUTXOTrie_ApplyState_CorruptedTrie(t *testing.T) {
+	db := mocks.NewTestDB()
+	trie := NewUTXOTrie(db.DB())
+
+	for i := 1; i < 24; i++ {
+		err := db.Update(func(txn *badger.Txn) error {
+			_, err := trie.ApplyState(txn, objs.TxVec{}, uint32(i))
+			return err
+		})
+		require.Nil(t, err)
+	}
+
+	// Height 24, tx 0
+	tx := &objs.Tx{
+		Fee:  uint256.Zero(),
+		Vin:  make([]*objs.TXIn, 1),
+		Vout: make([]*objs.TXOut, 21),
+	}
+
+	tx.Vin[0] = createVinTx(
+		t,
+		"01011b8f4e22353081e3870bde9171a6fd62924f6221fcb4cba1941896214fb210d92855a64346ca14a8bc9b343b649da07dd38dd94c3881ad305f3af91372ef5ea901",
```

Comment:
Try not to use magic addresses and numbers unless there's literally no other way to accomplish things. If that is the case, documentation should be provided for how those values came to be. If the values are pulled from a live environment they should be stored in a separate JSON file and read in as part of the test along with instructions for how to regenerate that file in the future. `Golden Files` is one of the terms for this, and [here](https://ieftimov.com/posts/testing-in-go-golden-files/) is a blog post with some additional context and some approaches (not the only approach, just one of).

Reply:
Some form of magic numbers are required.
This comes from the fact that a value in the SMT must first be added and then deleted,
and the additional values which are added must be in a particular form.

Reply:
I'm defining magic numbers as "literal values used in code without explanation of its meaning".

Using constant literal values is totally fine, but they need to be explained. This test as currently written is incomprehensible to anyone without extensive in depth knowledge of the internals of our implementation, making it impossible for new team members to help maintain the system.